internal/clients/http/azure/create_vm.go

```diff
@@ -0,0 +1,395 @@
+// strongly inspired by https://github.com/Azure-Samples/azure-sdk-for-go-samples/tree/main/sdk/resourcemanager/compute/create_vm
+// from commit 2ea51d9744d5d4680b4983b2e292a7b1d0085ff4
+
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/RHEnVision/provisioning-backend/internal/ptr"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+)
+
+var subscriptionId string
+
+const TraceName = "github.com/RHEnVision/provisioning-backend/internal/clients/http/azure"
+
+const (
+	resourceGroupName = "redhat-deployed2"
```

Comment:
Lighthouse supports attaching to either subscription or resource group. Are you testing this with subscription attachement or resource one? I am not sure if RGs can form a tree: https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/manage-resource-groups-portal

Reply:
I think we talked about this and agreed we would enforce Lighthouse on a resource group, just for cleaner environment, therefore this would probably not work I guess?!?

Reply:
Ah! I did not remember us deciding to force it on ResourceGroup, I've tested with attaching to sub, but with this, it looked clumsy, so I agree attaching to ResourceGroup makes more sense :)

Reply:
I wonder how are we going to enforce this, sources only stores UUID, do they? We won't be able to tell to which resource it was approved. We need to test this.

Reply:
The Lighthouse template allows you to chose and it will force CU to choose ResourceGroup if we choose to bind to ResourceGroup.

Reply:
Ugh, it actually will Create statically named resource group :face_exhaling: